File: app/src/main/java/ir/example/activity/SplashActivity.java
package ir.example.activity;

import android.content.Intent;
import android.os.Bundle;
import android.view.Window;
import android.view.WindowManager;
import ir.example.R;

public class SplashActivity extends BaseActivity {
    Thread splashTread;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setFullScreen();
        setContentView(R.layout.activity_splash);
        startSplash();
    }

    private void setFullScreen() {
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN,
                WindowManager.LayoutParams.FLAG_FULLSCREEN);
    }

    private void startSplash() {
        splashTread = new Thread() {
            @Override
            public void run() {
                try {
                    int waited = 0;
                    // Splash screen pause time
                    while (waited < 13000) {
                        sleep(100);
                        waited += 100;
                    }

                } catch (InterruptedException e) {
                    // do nothing
                } finally {
                    goToMain();
                }
            }
        };

        splashTread.start();
    }

    private void goToMain() {
        startActivity(new Intent(this, MainActivity.class));
    }

}

File: app/src/main/java/ir/example/activity/SplashActivity.kt
package ir.example.activity

import android.content.Intent
import android.os.Bundle
import android.view.Window

class SplashActivity : BaseActivity() {
    var splashThread: Thread?=null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setFullScreen()
        setContentView(ir.example.R.layout.activity_splash)
    }

    private fun setFullScreen() {
        requestWindowFeature(Window.FEATURE_NO_TITLE)
    }

    private fun goToMain() {
    startActivity(Intent(this,MainActivity::class.java))
    }

    private fun startSplash() {
        splashThread = object : Thread() {
            override fun run() {

                try {
                    var waited = 0
                    // Splash screen pause time
                    while (waited < 13000) {
                        Thread.sleep(100)
                        waited += 100
                    }

                } catch (e: InterruptedException) {
                    // do nothing
                } finally {
                    goToMain()
                }
            }
        }

        (splashThread as Thread).start()
    }
}